diff --git a/board/phytec/phycore_imx8mp/spl.c b/board/phytec/phycore_imx8mp/spl.c
index 3de1e11a..efff2c27 100644
--- a/board/phytec/phycore_imx8mp/spl.c
+++ b/board/phytec/phycore_imx8mp/spl.c
@@ -37,7 +37,28 @@ void set_dram_timings_2ghz_8gb(void);
 
 int spl_board_boot_device(enum boot_device boot_dev_spl)
 {
+#ifdef CONFIG_SPL_BOOTROM_SUPPORT
 	return BOOT_DEVICE_BOOTROM;
+#else
+	switch (boot_dev_spl) {
+	case SD1_BOOT:
+	case MMC1_BOOT:
+	case SD2_BOOT:
+	case MMC2_BOOT:
+		return BOOT_DEVICE_MMC1;
+	case SD3_BOOT:
+	case MMC3_BOOT:
+		return BOOT_DEVICE_MMC2;
+	case QSPI_BOOT:
+		return BOOT_DEVICE_NOR;
+	case NAND_BOOT:
+		return BOOT_DEVICE_NAND;
+	case USB_BOOT:
+		return BOOT_DEVICE_BOARD;
+	default:
+		return BOOT_DEVICE_NONE;
+	}
+#endif
 }
 
 enum phytec_imx8mp_ddr_eeprom_code {
@@ -227,3 +248,10 @@ void board_init_f(ulong dummy)
 	/* DDR initialization */
 	spl_dram_init();
 }
+
+  // Falcon mode AN13709 4.3.3
+#ifdef CONFIG_SPL_OS_BOOT
+int spl_start_uboot(void) {
+ return 0;
+}
+#endif
diff --git a/common/spl/spl.c b/common/spl/spl.c
index b452d4fe..3d554e38 100644
--- a/common/spl/spl.c
+++ b/common/spl/spl.c
@@ -36,9 +36,19 @@
 #include <fdt_support.h>
 #include <bootcount.h>
 #include <wdt.h>
+#include <asm/gpio.h>
+#include <linux/delay.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
+
+// Falcon mode AN13709 4.3.7
+// from u-boot-imx/arch/arm/dts/imx8mp-phyboard-pollux-rdk.dts:
+// MX8MP_IOMUXC_SAI1_MCLK__GPIO4_IO20
+#define RESET_ETH1_GPIO IMX_GPIO_NR(4,20)
+#define USDHC_GPIO_PAD_CTRL (PAD_CTL_HYS | PAD_CTL_DSE1)
+
+
 #ifndef CONFIG_SYS_UBOOT_START
 #define CONFIG_SYS_UBOOT_START	CONFIG_SYS_TEXT_BASE
 #endif
@@ -743,6 +753,15 @@ void board_init_r(gd_t *dummy1, ulong dummy2)
 		}
 	}
 
+
+		// Falcon mode AN13709 4.3.7
+	gpio_request(RESET_ETH1_GPIO, "reset_eth1_gpio");
+	gpio_direction_output(RESET_ETH1_GPIO, 0);
+	mdelay(1);
+	gpio_direction_output(RESET_ETH1_GPIO, 1);
+	mdelay(1);
+
+
 #if CONFIG_IS_ENABLED(BOARD_INIT)
 	spl_board_init();
 #endif
@@ -821,7 +840,10 @@ void board_init_r(gd_t *dummy1, ulong dummy2)
 		spl_fixup_fdt((void *)CONFIG_SYS_SPL_ARGS_ADDR);
 #endif
 		spl_board_prepare_for_linux();
-		jump_to_image_linux(&spl_image);
+		// jump_to_image_linux(&spl_image);
+		typedef void __noreturn (*image_entry_noargs_t)(void);  // Falcon mode AN13709 4.3.6
+		image_entry_noargs_t image_entry = (image_entry_noargs_t)0x00970000;
+		image_entry();
 #endif
 	default:
 		debug("Unsupported OS image.. Jumping nevertheless..\n");
diff --git a/common/spl/spl_legacy.c b/common/spl/spl_legacy.c
index 2ec71544..650349d4 100644
--- a/common/spl/spl_legacy.c
+++ b/common/spl/spl_legacy.c
@@ -39,8 +39,10 @@ int spl_parse_legacy_header(struct spl_image_info *spl_image,
 	} else {
 		spl_image->entry_point = image_get_ep(header);
 		/* Load including the header */
-		spl_image->load_addr = image_get_load(header) -
-			header_size;
+		// spl_image->load_addr = image_get_load(header) -
+		// 	header_size;
+		spl_image->load_addr = image_get_ep(header) -  // Falcon mode AN13709 4.3.4
+				header_size;
 		spl_image->size = image_get_data_size(header) +
 			header_size;
 	}
diff --git a/common/spl/spl_mmc.c b/common/spl/spl_mmc.c
index cbbd166e..bcec92ba 100644
--- a/common/spl/spl_mmc.c
+++ b/common/spl/spl_mmc.c
@@ -46,6 +46,9 @@ static int mmc_load_legacy(struct spl_image_info *spl_image,
 			  sector + image_offset_sectors,
 			  image_size_sectors,
 			  (void *)(ulong)spl_image->load_addr);
+	unsigned long count1 = blk_dread(mmc_get_blk_desc(mmc),  // Falcon mode AN13709 4.3.5
+				0x2FCF280, 0x100,                                  // ATF on SD card
+				(void*)(ulong)0x00970000);                         // write ATF from SD to RAM
 	debug("read %x sectors to %lx\n", image_size_sectors,
 	      spl_image->load_addr);
 	if (count != image_size_sectors)
diff --git a/configs/phycore-imx8mp_defconfig b/configs/phycore-imx8mp_defconfig
index 349b9ab6..2af849ef 100644
--- a/configs/phycore-imx8mp_defconfig
+++ b/configs/phycore-imx8mp_defconfig
@@ -35,7 +35,7 @@ CONFIG_BOOTCOMMAND="mmc dev ${mmcdev}; if mmc rescan; then env exists dofitboot
 CONFIG_DEFAULT_FDT_FILE="oftree"
 CONFIG_BOARD_LATE_INIT=y
 CONFIG_SPL_BOARD_INIT=y
-CONFIG_SPL_BOOTROM_SUPPORT=y
+# CONFIG_SPL_BOOTROM_SUPPORT is not set  // Falcon mode AN13709 4.3.1
 CONFIG_SPL_SYS_MALLOC_SIMPLE=y
 CONFIG_SPL_SEPARATE_BSS=y
 CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR=y
@@ -156,3 +156,4 @@ CONFIG_USB_GADGET_PRODUCT_NUM=0x0152
 CONFIG_IMX_WATCHDOG=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_FDT_FIXUP_PARTITIONS=y
+CONFIG_CMD_SPL=y
diff --git a/include/configs/phycore_imx8mp.h b/include/configs/phycore_imx8mp.h
index 1e801264..f7feece9 100644
--- a/include/configs/phycore_imx8mp.h
+++ b/include/configs/phycore_imx8mp.h
@@ -178,4 +178,20 @@
 #define CONFIG_USB_MAX_CONTROLLER_COUNT         2
 #define CONFIG_SERIAL_TAG
 
+  // Falcon mode AN13709 4.3.2
+#define CONFIG_CMD_SPL 1 // enable spl export command
+// #define CONFIG_SPL_MMC_SUPPORT 1 // for reading from MMC
+#define CONFIG_SPL_MMC 1 // for reading from MMC - PD23.1.0 - 5.15.71
+#define CONFIG_SPL_LEGACY_IMAGE_SUPPORT 1
+/* Falcon Mode */
+#define CONFIG_SPL_OS_BOOT 1 // activate Falcon Mode
+ /* (leave this line commented until you finish all the configurations) */
+// RAM FDT address
+#define CONFIG_SYS_SPL_ARGS_ADDR 0x48000000
+/* Falcon Mode - MMC support */
+#define CONFIG_SYS_MMCSD_RAW_MODE_ARGS_SECTOR 0x2FAF080    // FDT sector start
+#define CONFIG_SYS_MMCSD_RAW_MODE_ARGS_SECTORS 0x200       // FDT sectors count
+#define CONFIG_SYS_MMCSD_RAW_MODE_KERNEL_SECTOR 0x2FAF280  // Kernel Image sector start
+#define CONFIG_SYS_OS_BASE 0x1000000  // fake address - needed constant for compilation in Falcon mode
+
 #endif /* __PHYCORE_IMX8MP_H */
diff --git a/include/log.h b/include/log.h
index ce48d514..9a1d3b50 100644
--- a/include/log.h
+++ b/include/log.h
@@ -15,6 +15,8 @@
 #include <linux/bitops.h>
 #include <linux/list.h>
 
+// #define DEBUG
+
 struct cmd_tbl;
 
 /**
